src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,167 @@
+  [DrawerSizes.sm]: 360,
+  [DrawerSizes.md]: 480,
+  [DrawerSizes.lg]: 720,
+  [DrawerSizes.xl]: 960,
+};
+
+const DrawerWrapper = styled.div`
+  position: fixed;
+  height: 100%;
+  top: 0;
+  right: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  z-index: ${getDepth('modal')};
+`;
+
+const Header = styled(Padbox)`
+  border-bottom: 1px solid ${getColor('neutral.500')};
+`;
+
+const BaseDrawer = styled.div<{ $maxWidth: number }>`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  width: 100%;
+  max-width: ${({ $maxWidth }) => pxToRem($maxWidth)};
+  background-color: ${getColor('neutral.0')};
+  box-shadow: -4px 0px 16px rgba(0, 0, 0, 0.1);
+`;
+
+const TitleWrapper = styled(Padbox)`
+  overflow: hidden;
+`;
+
+/* stylelint-disable */
+const Title = styled(H4)`
+  margin: 0;
+  display: -webkit-box;
+  -webkit-line-clamp: 2;
+  -webkit-box-orient: vertical;
+`;
+/* stylelint-enable */
+
+const Content = styled(Padbox)`
+  overflow-y: auto;
+  flex-grow: 2;
+`;
+
+const Footer = styled(Padbox)`
+  border-top: 1px solid ${getColor('neutral.500')};
+`;
+
+const Adornment = styled(Padbox)`
+  max-width: 36px;
+  max-height: 36px;
+`;
+
+const Drawer = forwardRef<HTMLDivElement, DrawerProps>(
+  (
+    {
+      title,
+      onClose,
+      footer,
+      size = DrawerSizes.md,
+      adornment,
+      hasBackdrop = true,
+      children,
+      ...props
+    },
+    ref,
+  ) => {
+    const { portalsContainerId } = useContext(DSContext);
+    const { Portal } = usePortal({
+      containerId: portalsContainerId,
+      internalShowHide: false,
+      autoRemoveContainer: false,
+    });
+    const hasFooter = isNotUndefined(footer);
+    const hasAdornment = isNotUndefined(adornment);
+
+    const drawerRef = useOuterClick<HTMLDivElement>(onClose);
+
+    useLockBodyScroll({ enabled: hasBackdrop });
+    const DrawerBox = () => (
+      <BaseDrawer
+        ref={mergeRefs<HTMLDivElement>(drawerRef, ref)}
+        $maxWidth={widthVariants[size]}
+        {...props}
+      >
+        <Header>
+          <Inline stretch={StretchEnum.start}>
+            <Padbox paddingSize={SpaceSizes.lg} paddingType="squish">
+              <Inline align="center" gap={SpaceSizes.md}>
+                {hasAdornment && <Adornment>{adornment}</Adornment>}
+                <TitleWrapper>
+                  <Title>{title}</Title>
+                </TitleWrapper>
+              </Inline>
+            </Padbox>
+            <CloseButton
+              marginCompensation={SpaceSizes.none}
+              onClose={onClose}
+            />
+          </Inline>
+        </Header>
+        <Content
+          paddingSize={SpaceSizes.lg}
+          paddingType={hasFooter ? 'squish' : 'square'}
+        >
+          {children}
+        </Content>
+        {hasFooter && (
+          <Footer paddingSize={SpaceSizes.lg} paddingType="squish">
+            {footer}
+          </Footer>
+        )}
+      </BaseDrawer>
+    );
```

Comment:
This `DrawerBox` component should live outside of `Drawer` component. If we keep in inside this component will be initialized on every rerender. So we need to extract it and pass needed properties inside `DrawerBox`.